Sources/CodeScanner/CodeScanner.swift

```diff
@@ -216,12 +238,26 @@ public struct CodeScannerView: UIViewControllerRepresentable {
             previewLayer.frame = view.layer.bounds
             previewLayer.videoGravity = .resizeAspectFill
             view.layer.addSublayer(previewLayer)
+            addviewfinder()
 
             if (captureSession?.isRunning == false) {
                 captureSession.startRunning()
             }
         }
 
+        private func addviewfinder() {
+            if showViewfinder,
+               let imageView = viewFinder {
+                view.addSubview(imageView)
+                NSLayoutConstraint.activate([
+                    imageView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+                    imageView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+                    imageView.widthAnchor.constraint(equalToConstant: 200),
+                    imageView.heightAnchor.constraint(equalToConstant: 200),
+                ])
+            }
+        }
```

Comment:
Can I suggest using a guard statement here to return early? This will detangle the function a bit.

```suggestion
        private func addviewfinder() {
            guard showViewfinder, let imageView = viewFinder else { return }
            
            view.addSubview(imageView)
            NSLayoutConstraint.activate([
                imageView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
                imageView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
                imageView.widthAnchor.constraint(equalToConstant: 200),
                imageView.heightAnchor.constraint(equalToConstant: 200),
            ])
        }
```

Reply:
Makes sense, great suggestion 👍